osc-server/src/main/java/org/osc/core/broker/job/Graph.java

```diff
@@ -162,25 +162,37 @@ private Graph(Set<T> nodes, Multimap<T, T> edges) {
         this.edges = edges;
     }
 
-    public Set<Edge<T>> getEdges() {
+    private synchronized Set<Edge<T>> getEdges() {
         Set<Edge<T>> edgeSet = new HashSet<Edge<T>>();
-        for (Map.Entry<T, T> edge : this.edges.entries()) {
+
+        Set<Map.Entry<T, T>> copy;
```

Comment:
you dont neeed this since the method is already synchronized

Reply:
I actually thought of removing synchronized.